From 185cb4d8050395af12da09b64cf52d31877811e5 Mon Sep 17 00:00:00 2001
From: Stuart Yoder <stuart.yoder@arm.com>
Date: Mon, 24 May 2021 17:21:49 -0500
Subject: [PATCH 04/37] uefi-sct/SctPkg: sign the .efi images

sign all the .efi images used by SCT so they can be loaded
when Secure Boot is enabled

Signed-off-by: Stuart Yoder <stuart.yoder@arm.com>
---
 uefi-sct/SctPkg/CommonGenFramework.sh | 46 +++++++++++++++++++++++++++
 uefi-sct/SctPkg/UEFI/UEFI_SCT.dsc     |  4 +++
 2 files changed, 50 insertions(+)

diff --git a/uefi-sct/SctPkg/CommonGenFramework.sh b/uefi-sct/SctPkg/CommonGenFramework.sh
index 8d58bed9..7b8ff473 100755
--- a/uefi-sct/SctPkg/CommonGenFramework.sh
+++ b/uefi-sct/SctPkg/CommonGenFramework.sh
@@ -94,6 +94,7 @@ CopyDependency()
     ls -h $ProcessorType/$1_*.ini   >> temp.txt 2>NUL
     ls -h $ProcessorType/$1_*.cmp   >> temp.txt 2>NUL
     ls -h $ProcessorType/$1_*.ucmp  >> temp.txt 2>NUL
+    ls -h $ProcessorType/$1_*.der  >> temp.txt 2>NUL
 
     while read line 
       do 
@@ -102,6 +103,30 @@ CopyDependency()
     rm -f temp.txt >NUL
 }
 
+# *********************************************
+# sign .efi executables for Secure Boot
+# *********************************************
+
+SecureBootSign()
+{
+for f in $1/*.efi
+do
+        echo "sbsign --key $ProcessorType/SecureBoot_TestDB1.key --cert $ProcessorType/SecureBoot_TestDB1.crt $f --output $f"
+        sbsign --key $ProcessorType/SecureBoot_TestDB1.key --cert $ProcessorType/SecureBoot_TestDB1.crt $f --output $f
+done
+
+}
+
+SecureBootSignDependency()
+{
+for f in $Framework/Dependency/$1BBTest/*.efi
+do
+        echo "sbsign --key $ProcessorType/SecureBoot_TestDB1.key --cert $ProcessorType/SecureBoot_TestDB1.crt $f --output $f"
+        sbsign --key $ProcessorType/SecureBoot_TestDB1.key --cert $ProcessorType/SecureBoot_TestDB1.crt $f --output $f
+done
+
+}
+
 # *********************************************
 # For UEFI SCT
 # *********************************************
@@ -268,6 +293,27 @@ then
     CopyDependency PciRootBridgeIo
     CopyDependency PxeBaseCode
     CopyDependency ConfigKeywordHandler
+    CopyDependency SecureBoot
+
+    # *********************************************
+    # Sign the .efi executables for use with Secure Boot
+    # *********************************************
+
+    SecureBootSign $Framework
+    SecureBootSign $Framework/Support
+    SecureBootSign SctPackage$ProcessorType
+    SecureBootSign $Framework/SCRT
+    SecureBootSign $Framework/Test
+    SecureBootSign $Framework/Ents/Support
+    SecureBootSign $Framework/Ents/Test
+
+    SecureBootSignDependency LoadedImage
+    SecureBootSignDependency ImageServices
+    SecureBootSignDependency ProtocolHandlerServices
+    SecureBootSignDependency ConfigKeywordHandler
+    SecureBootSignDependency Ebc
+    SecureBootSignDependency PciIo
+
 fi
 
 # *********************************************
diff --git a/uefi-sct/SctPkg/UEFI/UEFI_SCT.dsc b/uefi-sct/SctPkg/UEFI/UEFI_SCT.dsc
index aa54b6dc..5513ce21 100644
--- a/uefi-sct/SctPkg/UEFI/UEFI_SCT.dsc
+++ b/uefi-sct/SctPkg/UEFI/UEFI_SCT.dsc
@@ -422,6 +422,10 @@ SctPkg/TestCase/UEFI/EFI/Protocol/PxeBaseCode/BlackBoxTest/Dependency/Config/Con
 # Dependency files for Config Keyword Handler Protocol Test
 #
 SctPkg/TestCase/UEFI/EFI/Protocol/ConfigKeywordHandler/BlackBoxTest/Dependency/SampleDriver/DriverSampleDxe.inf
+#
+# Dependency files for Secure Boot Test
+#
+SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Keys/Keys.inf
 
 #
 # Support Files
-- 
2.17.1

